### PR TITLE
Set ITT outcome endpoint

### PR DESCRIPTION
**Pull request by gunndabad**
### Context

https://trello.com/c/ngFfEsOx/16-register-api-create-put-request-for-qts

### Changes proposed in this pull request

Adds the 'Set ITT outcome' endpoint for Register. Given a TRN, an outcome and an UKPRN find the matching Person + QTS records and update their status.

Part of this adds support for error codes so our API consumers have a nicer way of catching errors without string matching.

### Guidance to review

Best reviewed commit-by-commit.

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

File: docs/api-specs/v2.json [hunk 1]
```diff
@@ -5,6 +5,100 @@
...
+      "put": {
+        "tags": [
+          "IttOutcome"
+        ],
+        "summary": "Sets ITT outcome for a teacher",
+        "parameters": [
+          {
+            "name": "trn",
+            "in": "path",
+            "description": "The TRN of the teacher to set ITT outcome for.",
```
Comment (dankmitchell) [inline docs/api-specs/v2.json:18]: @benilovj should we go with `trainee` at this point, rather than `teacher`?

Comment (dankmitchell) [reply docs/api-specs/v2.json:18]: hmm actually, in the context of DQT - every record is a teacher even though we know that's not correct so let's leave as is.

**Review by MrKevJoy — APPROVED** [review]

File: src/DqtApi/V2/Handlers/SetIttOutcomeHandler.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            _dataverseAdapter = dataverseAdapter;
+        }
+
+        public async Task<SetIttOutcomeResponse> Handle(SetIttOutcomeRequest request, CancellationToken cancellationToken)
+        {
+            var teachers = (await _dataverseAdapter.GetTeachersByTrn(request.Trn, activeOnly: true)).ToArray();
+
+            if (teachers.Length == 0)
+            {
+                throw new ErrorException(10001);  // Teacher with specified TRN not found
```
Comment (defong) [inline src/DqtApi/V2/Handlers/SetIttOutcomeHandler.cs:29]: It feels like the error numbers more like 

```csharp
ErrorCode.TeacherWithSpecifiedTRNNotFound
```

Comment (gunndabad) [reply src/DqtApi/V2/Handlers/SetIttOutcomeHandler.cs:29]: @defong I like that. I'll change them up.

**Pull request merged**